File: code/regex/lua/regex.lua
#!/usr/bin/env lua
local i = 0
r1 = ""
r2 = ""

while i < 29 do
 i = i+1
 r1 = r1 .. "a?"
 r2 = r2 .."a"
 r = r1 .. r2
 if string.match(r2, r) then
   print (r2 .. " matches " .. r)
 else
   print ("matchin failure")
 end
end

File: code/regex/ruby/regex.rb
#!/usr/bin/env ruby

regex_part = ""
string = ""

29.times do
  regex_part += "a?"
  string += "a"

  regex = "#{regex_part}#{string}"
  if string.match?(regex)
    puts "#{string} matches #{regex}"
  else
    puts "matching failure"
  end
end
